packages/agents/src/index.ts

```diff
@@ -776,6 +908,80 @@ export class Agent<Env, State = unknown> extends Server<Env> {
     // biome-ignore lint/complexity/noBannedTypes: <explanation>
     return callableMetadata.has(this[method as keyof this] as Function);
   }
+
+  /**
+   * Connect to a new MCP Server
+   *
+   * @param url MCP Server SSE URL
+   * @param callbackHost Base host for the agent, used for the redirect URI.
+   * @param agentsPrefix agents routing prefix if not using `agents`
+   * @returns authUrl
+   */
+  async addMcpServer(
+    url: string,
+    callbackHost: string,
+    agentsPrefix = "agents"
```

Comment:
- [ ] Add additional MCP connect parameters here

Reply:
I decided to:
* Remove client capabilities options (because I realized they're redundant with the client options type)
* Prevent the user from configuring transport options from this API because it might change once we add support for the streamable transport.

Reply:
Actually refined this a bit more. Now:
* we allow headers as transport options (because of #253)
* persist options in sqlite storage so we recreate clients with the same options upon reconnection